src/main/java/com/minecolonies/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -21,520 +21,550 @@
+             requestSeeds = false;
+             //Flag 1+2 is needed for updates
+             return true;
+         }
+         else
+         {
+             return false;
+         }
+     }
+
+     /**
+      * Terminates the planting process and resets the task.
+      *
+      * @param buildingFarmer the building of the farmer.
+      * @param field          the field being planted.
+      * @return the next state.
+      */
+     @NotNull
+     private AIState terminatePlanting(@NotNull BuildingFarmer buildingFarmer, @NotNull Field field)
+     {
+         if (requestSeeds)
+         {
+             chatSpamFilter.talkWithoutSpam("entity.farmer.NeedSeed", field.getSeed().getItemStackDisplayName(new ItemStack(field.getSeed())));
+         }
+         else
+         {
+             buildingFarmer.getCurrentField().setNeedsWork(false);
+             buildingFarmer.getCurrentField().setFieldStage(PLANTED);
+         }
+
+         resetVariables();
+         return AIState.IDLE;
+     }
+
+     /**
+      * Executes the harvesting of the field.
+      *
+      * @return the next state.
+      */
+     @NotNull
+     private AIState harvest()
+     {
+         @Nullable BuildingFarmer buildingFarmer = getOwnBuilding();
+
+         if (buildingFarmer == null || buildingFarmer.getCurrentField() == null)
+         {
+             return AIState.PREPARING;
+         }
+         @Nullable Field field = getOwnBuilding().getCurrentField();
+
+         if (workingOffset != null)
+         {
+             BlockPos position = field.getLocation().down().south(workingOffset.getZ()).east(workingOffset.getX());
+             if (walkToBlock(position.up()))
+             {
+                 return AIState.FARMER_HARVEST;
+             }
+
+             if (shouldHarvest(position))
+             {
+                 worker.addExperience(0.5);
+                 mineBlock(position.up());
+             }
+         }
+
+         if (!handleOffset(field))
+         {
+             buildingFarmer.getCurrentField().setNeedsWork(false);
+             buildingFarmer.getCurrentField().setFieldStage(EMPTY);
+             return AIState.IDLE;
+         }
+         BlockPos position = field.getLocation().down().south(workingOffset.getZ()).east(workingOffset.getX());
+         if (shouldHarvest(position))
+         {
+             mineBlock(position.up());
+         }
+
+         setDelay(workingDelay);
+         return AIState.FARMER_HARVEST;
+     }
+
+     /**
+      * Checks if the crop should be harvested.
+      *
+      * @param position the position to check.
+      * @return true if should be hoed.
+      */
+     private boolean shouldHarvest(@NotNull BlockPos position)
+     {
+         IBlockState state = world.getBlockState(position.up());
+
+         if (state.getBlock() instanceof IGrowable && state.getBlock() instanceof BlockCrops)
+         {
+             @NotNull BlockCrops block = (BlockCrops) state.getBlock();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'block' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)